lib/DynamicSelectionFilter/DynamicSelectionFilter.js

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import { Loading } from '@folio/stripes/components';
+import { useOkapiKy } from '@folio/stripes/core';
+
+import { buildFiltersObj } from '../AcqList/utils';
+import { DynamicSelection } from '../DynamicSelection';
+import { FilterAccordion } from '../FilterAccordion';
+
+export const DynamicSelectionFilter = ({
+  activeFilters,
+  api,
+  closedByDefault,
+  dataFormatter,
+  disabled,
+  id,
+  labelId,
+  name,
+  onChange,
+  queryBuilder,
+  valueKey = 'id',
+  ...rest
+}) => {
+  const { search } = useLocation();
+  const ky = useOkapiKy();
+  const [isLoading, setIsLoading] = useState();
+  const [initialOptions, setInitialOptions] = useState([]);
+
+  const onSelectionChange = useCallback(
+    val => onChange({ name, values: [val] }),
+    [name, onChange],
+  );
+
+  useEffect(() => {
+    const initialFilterValue = buildFiltersObj(search)[name]?.[0];
+
+    if (initialFilterValue) {
+      setIsLoading(true);
+
+      const searchParams = {
+        query: `${valueKey}==${initialFilterValue}`,
+      };
+
+      ky.get(api, { searchParams })
+        .json()
+        .then(dataFormatter)
+        .then(setInitialOptions)
+        .then(() => setIsLoading(false));
```

Comment:
the same, if erorr - user'll see infinite loading. Plz add `.catch` and `.finaly(setIsLoading)`

Reply:
Done. Thanks!)